docs/design_proposals/design-proposal-template.md

```diff
@@ -0,0 +1,83 @@
+# Title
+
+* Author(s): \<your name\>
+* Design Shepherd: As mentioned in the document here.
+* Date: \<date\>
+
+## Background
+
+In this section, please mention and describe the new feature, re-design
+or re-factor.
+
+Please provide an rationale covering following points:
+
+1. Why is this required?
+2. If its re-design, What are cons with current implementation?
+3. Are there any another work-around and if yes, why not keep using it.
+
+Here is an example snippet for a new feature:
+
+___
+Currently, skaffold config supports `artifact.sync` as a way to sync files
+directly to pods. So far, artifact sync requires a specification of sync
+patterns like
+
+```yaml
+sync:
+  '*.js': app/
+```
+
+This is error prone and unnecessarily hard to use, because the destination is
+already contained in the Dockerfile for docker build. (see #1166, #1581).
+In addition, the syncing needs to handle special cases for globbing and often
+requires a long list of sync patterns (#1807)
+___
+
+## Design
+
+Please describe your solution. Please list any:
+
+* new config changes
+* interface changes
+* design assumptions
+
+For a new config change, please mention:
+
+* If its a backward compatible config change
+* If its a backward compatible config change, is there a migration path?
```

Comment:
I do not fully understand what is meant by the migration path. I presume, you do not mean the automatic schema upgrade? Can you elaborate on this one a bit more?

Reply:
thanks. i changed to mention how you are going to deprecate it with links to deprecation policy docs.